src/components/toolbar/type-selector/TypeSelector.tsx

```diff
@@ -129,6 +129,13 @@ export const TypeSelector: React.FC<{
 }> = ({ state, dispatch, view }) => {
   const options = buildOptions(state)
   const isInBody = hasParentNodeOfType(schema.nodes.body)(state.selection)
+  const $from = state.selection.$from
+  const sectionDepth = ($from.depth || 1) - 1
+  const parentSection = $from.node(sectionDepth)
+  const hasValidParent =
+    parentSection.type === schema.nodes.body ||
+    parentSection.type === schema.nodes.section
+
```

Comment:
I added this to disable it for paragraphs within [figure_element](https://github.com/Atypon-OpenSource/manuscripts-transform/blob/master/src/schema/nodes/figure_element.ts#L33)